### DISCUSSION TITLE
Node 14 support?

### DISCUSSION THREAD

**Discussion by borekb** [vercel/vercel#4754] [General] [answered]
Is there a rough idea where Node v14 support is coming to Vercel? Is the usual policy to wait for LTS (October) or does it usually happen sooner?

We'd like to support non-en-US locales of `Intl.NumberFormat` and those are only enabled on Node 13 & 14.

**Comment by mcsdevv** [comment]
Hey @borekb, thanks for raising this!

As soon as it's supported by our upstream providers, we'll look to implement it.

**Reply by borekb** [reply]
Ah, OK, makes sense. Just looking when AWS released Node 12 and "unfortunately" it was after it LTS'd, in November last year. So there's probably still some wait ahead.

**Reply by cyrus-za** [reply]
LTS is official today on Node but I cant find any indication from AWS as to when they will support it. If I do, I will post here. If everyone else could do the same, that'd be great. Based on Node 12 LTS last year, it seems like 2 week lag from AWS so should hopefully have support from AWS 2 weeks from now and hopefully vercel will support it shortly after

**Reply by damianobarbati** [reply]
So we're waiting aws to support node 14 on its lambdas, right?

**Reply by paulogdm** [reply]
That is correct @damianobarbati .

**Reply by mikaelgson** [reply]
https://aws.amazon.com/blogs/compute/node-js-14-x-runtime-now-available-in-aws-lambda/ 🎉 

Any estimate on how long for Vercel to enable support now that it’s available on AWS?

**Reply by mikestopcontinues** [reply]
@mcsdevv It looks like AWS just got 14.x.

**Reply by paulogdm** [reply]
Yep! We are working on it as we speak!

**Comment by matamatanot** [comment]
@mcsdevv 
https://github.com/nodejs/Release#release-schedule
Node 14.x start active LTS 2020-10-27 (Soon!).

Can you give some follow-up information about Node 14 support?

**Comment by brianespinosa** [comment]
@matamatanot according to AWS docs, it looks like a 14.x lambda is available yet. I am guessing that once it becomes available here, it can be be enabled with Vercel. I do not work for Vercel though, so this is just a guess. But at least there's a place to check back on for anyone landing here wondering what the status is.

https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html

**Reply by cyrus-za** [reply]
yep. I check it every 2nd or 3rd day. I also search for other sources in case they announce it somewhere before they update their docs. Nothing yet. Last year they supported v12 around 2 weeks after it became LTS. Its been almost 2 weeks since v14 became LTS. Hopefully within the next few days we get clarity

**Reply by cyrus-za** [reply]
Still nothing, and are we now facing a bug with 12.19 (which vercel upgraded to) due to breaks in undici. 
https://stackoverflow.com/questions/64293094/error-assertion-args3-isint32-failed

There is no way to force vercel to use 12.18 where this used to work. Very bad design to force a specific version

**Comment by ballonura** [comment]
Any news for Vercel's Node.js environment version upgrade to 14.x.x ? (Netlify already support)

From my last Netlify build:
![image](https://user-images.githubusercontent.com/22234568/99695122-de1f0280-2a95-11eb-9a0b-79d261300929.png)

According to [AWS Lambda runtimes](https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html), they still didnt support 14.x.x.

**Reply by brianespinosa** [reply]
There's no news until AWS supports it. That is what we are all waiting for.

**Reply by cyrus-za** [reply]
What does netlify use in order to support it? I was under the impression they also use AWS lambda and Google Cloud Functions

**Reply by ballonura** [reply]
My fault, Netlify still **didn't support** Node.js version 14.x.x, their last lambda node version update (to 12.x.x) was on December, 14, 2019, as they mentioned here:
[Node.js 12 now available for Netlify Functions](https://www.netlify.com/blog/2019/12/04/node.js-12-now-available-for-netlify-functions/)

**Reply by paulogdm** [reply]
As soon as AWS releases support for it, we will give our users the ability to deploy Node 14. Can't wait for it as well!

**Reply by cyrus-za** [reply]
Its been over 2 months since 14 became LTS, yet no word from AWS. Does anyone know what the story is there?

**Reply by brianespinosa** [reply]
Ask AWS.

**Reply by cyrus-za** [reply]
I did. No response

**Comment by carlosfaria94** [comment]
@mcsdevv I understand that you need to keep compatibility with upstream dependencies. However, to build a web application it would be better to have Nodejs 14 and 15 with NPM 7 (https://github.blog/2020-10-13-presenting-v7-0-0-of-the-npm-cli/) to drastically improve the build time. It will be a win-win to Vercel and developers.

For instance, Netlify only supports NodeJs 10 and 12 to run serverless functions, but I can easily use any nodejs version during a building process, they use nvm for that.

Because during a build process, what really matters is **speed** and **flexibility** of choosing which version, in order to get static files to be delivered to a CDN. Running serverless functions is a complete different problem, that needs a compatible and well-tested runtime.

**Reply by mcsdevv** [reply]
Hey @carlosfaria94, we've been looking into that recently for this very reason - if we make that change we'll make sure it's available on the changelog.

**Reply by cyrus-za** [reply]
@mcsdevv any news on this?

**Reply by paulogdm** [reply]
@cyrus-za Let's say it is a question of weeks now. We will get back to you very soon!

**Reply by cyrus-za** [reply]
Thanks @paulogdm I see its finally out

**Comment by paulogdm** [comment, marked as answer]
Hello everyone!

Global update: As you may already know, Node 14 was released for the AWS Lambda product. You can read their announcement [here](https://aws.amazon.com/blogs/compute/node-js-14-x-runtime-now-available-in-aws-lambda/). We should be ready to release it to Vercel in a few hours. Stay tuned here for updates.

---

### **EDIT: We announced it! https://twitter.com/vercel_changes/status/1357463723431100418**

**Reply by styfle** [reply]
Vercel announcement here: https://vercel.com/changelog/node-js-14-lts-is-now-available

**Reply by redstrike** [reply]
@styfle I got this error `NO_RESPONSE_FROM_FUNCTION` when executing the final build result (setting tsconfig.json to target `ES2020` and module `ES2020` with Node 14 LTS runtime). Is this a bug? I have to downgrade my tsconfig to make it work again.

**Reply by cyrus-za** [reply]
Thanks for the updates everyone! Going to try it out asap

**Reply by cyrus-za** [reply]
@styfle @paulogdm 
<img width="1026" alt="image" src="https://user-images.githubusercontent.com/1845861/107243500-bc4e6f80-6a35-11eb-9bea-90ff178af8c2.png">

**Reply by paulogdm** [reply]
@cyrus-za Make sure you select "Node 14" in your project: https://vercel.com/changelog/node-js-14-lts-is-now-available

**Reply by styfle** [reply]
@cyrus-za Are you using `builds` in vercel.json? Try removing it or make sure to use `@vercel` instead of `@now`.

**Reply by cyrus-za** [reply]
We are using builds. It's a server still using the node-server. I'll try switching it from @now/node-server to @vercel/node-server
I thought they were just aliases. Is @now referencing and outdated version?

**Reply by paulogdm** [reply]
@cyrus-za `@now/node-server` or `@vercel/node-server` has been deprecated. We recommend the usage of `@vercel/node`, which offers the same functionalities. No change in your code is required.
